### DISCUSSION TITLE
How to handle mimeopen prompt (shell-wait and shell-async)

### DISCUSSION THREAD

**Discussion by sarmong** [gokcehan/lf#1028] [Q&A] [answered]
I use `mimeopen` instead of `xdg-open`. If default app for this mimetype is not set, it prompts you to choose one.
![image](https://user-images.githubusercontent.com/42828375/204132258-cc4fc060-33a6-43b0-9b53-c3f0b310f59a.png)

I want to combine shell-wait and shell-async capabilities.

Expected behaviour: you run open command, get a prompt, select an app and then app opens, but lf disowns it and returns to the directory view.

I could not get the last part working with any combination of `& disown` and `setsid`. After you select the application, lf prompt just hangs
![image](https://user-images.githubusercontent.com/42828375/204132396-735a560a-a18e-4e47-b56a-d75330758b24.png)

**Comment by Aqa-Ib** [comment]
I am trying to solve this, did you solve it somehow?
It also happens when using "mimeopen --ask" option.

**Comment by jpggithub** [comment]
An issue has been put on https://github.com/mbeijen/File-MimeInfo/issues/53

**Comment by jpggithub** [comment]
I put a workaround on https://github.com/mbeijen/File-MimeInfo/issues/53

**Comment by joelim-work** [comment]
Thank you for investigating this. I agree the root cause is due to the design of `mimeopen` - the selected program should run as a separate process that is disassociated with the current terminal.

Because `lf` integrates with [a number of programs](https://github.com/gokcehan/lf/wiki/Integrations), I guess it is inevitable that there will be many issues raised with external programs here, even if the issue is not to do with `lf` itself.

**Comment by jpggithub** [comment, marked as answer]
Thanks @joelim-work for the reminder.
The good news is that  the maintainer of mimeopen has resolved the bug with a more elegant way than mine!
